### PR TITLE
Reload database when extracting database over it

### LINKED ISSUE #703: 'python3.7 killed by SIGABRT' when extracting ~/.var/app/com.borgbase.Vorta
**Describe the bug**
When extracing an archive to a new installation via Vorta (not via command line), Vorta UI disappears and Fedora 33 reports a "python3.7 killed by SIGABRT" error.

Once this happens, the list of archives in the repository is shortened, by one (latest is lost) and you need to disconnect and reattach the repository (or do a 'Refresh').

My goal was to restore a backup to a new PC/installation (old one died), and inherit all my previous Vorta settings.  

Most likely Vorta doesn't like its own data being changed while it is currently running, which makes it that the user will always have to remember to exclude the ~/.var/app/com.borgbase.Vorta directory when doing an extract/restore of ~/.var directory.

**To Reproduce**
1. Run Vorta.
2. Go to the 'Archives' tab.
3. Highlight the latest archive.
4. Press the 'Extract' button.
5. Make sure the home/-USER-/.var.app/com.borgbase.Vorta directory is selected for extracting.

**Desktop (please complete the following information):**
- OS: Fedora 33 Workstation
- Vorta version: 0.7.1
- Installed from: Fedora Software (Flatpak)

Vorta and Borg versions can be found in Main Window > Misc Tab

**Additional context**
If appropriate include logs. Can be found in Main Window > Misc Tab > Log.

[xxx@localhost ~]$ flatpak info com.borgbase.Vorta

Vorta - Backup client

          ID: com.borgbase.Vorta
         Ref: app/com.borgbase.Vorta/x86_64/stable
        Arch: x86_64
      Branch: stable
     Version: v0.7.1
     License: GPL-3.0
      Origin: flathub
  Collection: org.flathub.Stable
Installation: system
   Installed: 54.0 MB
     Runtime: org.kde.Platform/x86_64/5.14
         Sdk: org.kde.Sdk/x86_64/5.14

      Commit: 96a3dd02990a4c03409e3ae06192951b77d4020bfbd3c04a405ab275fdb9c10e
      Parent: 109123847216e7a1a7e7d192f49feb75fb08962a516dad6fe70ea26bae8318ec
     Subject: Add --filesystem=~/.var/app/ to complete home dir (#30) (f301b4dc)
        Date: 2020-09-14 22:49:28 +0000
[xxx@localhost ~]$ python --version
Python 3.9.0

### PR DESCRIPTION
**Pull request by samuel-w**
Still clears archivemodel for some reason, need to fix that.
Edit: It happens only sometimes. Probably race condition.

Fixes #703

### CODE REVIEW

**Comment by samuel-w** [comment]
No race condition after all, it was caused by the bottom archive being selected which doesn't have the archive model data, creating the appearance that the archivemodel got cleared.

**Comment by samuel-w** [comment]
There might be more weird cases for extracting a db that:
Doesn't have the current profile
Doesn't have the current repo
Has different settings/schedule/global settings

Do we have a reload everything function we can call?

**Comment by samuel-w** [comment]
I think I can do this differently through renaming the db on restarting Vorta.

**Comment by samuel-w** [comment]
Since disagreement == too heated now, are you going to merge this or should I just not bother?
This kind of stuff is why I left as a maintainer.

**Comment by m3nu** [comment]
I wont merge this because:

- it runs for every single line of logs, but is rarely needed. Especially for `extract` it’s critical to keep it fast.
- the issue of overwriting files that are currently in use is common to pretty much every other app. You cant filter for every single possibility.

Sorry if you did the work for nothing or disagree with the above. As mentioned in the contributor docs, it’s best to discuss new features before actually coding them.

I realize that choosing features based on incomplete information will be opinionated. I dont have a better way right now. Using Github Discussions with the upvote feature is an attempt to make it more democratic.

**Comment by samuel-w** [comment]
I don't find a crash during an operation (extract) acceptable.
This only runs for extract commands only.

What solution do you propose to prevent crashing when extracting settings.db?

**Comment by samuel-w** [comment]
Also, during an extract, most time is taken by disk IO.

**Comment by samuel-w** [comment]
I did an experiment, comparing speed with and without.

Setup:
200k random files, 1k size each.
Extracted to /tmp,  to increase IO speed.
87.15 seconds with this code, 88.04 seconds without.

My conclusion is that there is very little different one if statement will do to extract.

**Review by m3nu — COMMENTED** [review]
> I don't find a crash during an operation (extract) acceptable.
> This only runs for extract commands only, filtering out for only settings.db. As long as settings.db is called settings.db there is no problem.
> 
> What solution do you propose to prevent crashing when extracting settings.db?

Just don't overwrite the files of an active program. 😄 Can you remove your Chrome/Firefox profile while using it? Why would this be a feature?

**Comment by m3nu** [comment]
@samuel-w , I'm about to merge #898. Please help to review, if possible. This is the feature with the most feedback and most usage.

**Comment by samuel-w** [comment]
For a web browser that only really writes to a downloads folder, yes. But for a backup program that backs up all user data, if you include the backup program's settings, no.

File: src/vorta/models.py [hunk 1]
```diff
@@ -252,6 +253,12 @@ def get_misc_settings():
     return settings
 
 
+def connect_db():
```
[the remark this answers is not in the document]
Comment (samuel-w) [reply src/vorta/models.py:275]: I don't think I can, because of the con.

[the remark this answers is not in the document]
Comment (samuel-w) [reply src/vorta/models.py:275]: Yeah, I tried to remove it and it broke the tests. I could try to change the test code, but based on past experience changing it makes things worse.

**Comment by m3nu** [comment]
> I backup my entire home directory (all of my data).
> I restore my home directory (all of my data).
> Vorta crashes.

We can't support this because you can't do a restore of your home folder **while logged in**. You need to log in as root, restore `/home/samuel`, maybe fix permissions and then log in as `samuel` again.

I'm sure you got it working for Vorta's DB, but what if users overwrite other files they have open?

**Comment by samuel-w** [comment]
If restoring a home folder, its typically from a blank slate, with no other programs open other than maybe a web browser and terminal. Firefox actually handles this well, they put each profile in a folder with a random name to prevent crashing.

I could put a dialog recommending to restart after restoring.

**Comment by m3nu** [comment]
Restarting isn't enough. As long as you have a desktop env going, files will be used. The steps to restore a full home folder are:

1. Log out
2. Log in as root or some other admin user
3. Restore other user's home folder. Can be done with Vorta.
4. Fix permissions if needed. macOS will do this automatically if a new user is created that already has a home folder (e.g. after restore)
5. Log in as restored user

If you think I'm lying, please get a second opinion. We won't encourage anyone to overwrite their home folder, while logged in.

Restoring may work for a simple Linux desktop without much stuff open, but it will cause problems on macOS for sure.

Even if Firefox uses a random profile name, the backup you did before will still use the same name. They don't do it for that.

**Comment by samuel-w** [comment]
The original issue was done in Linux.
I don't want to encourage overwriting the home folder either and this doesn't do it.
In the end, all I want to prevent is the crash from overwriting settings.db. 
It might not even come from a full restore, since sometime I mess up my settings and need to restore my settings.db

**Comment by m3nu** [comment]
Sounds like this mostly solves your own problem. Problem is that eventually you will move on and then I'm stuck maintaining your code that doesn't serve anyone and that I can't remember why it's there.

Myself, I sometimes rename `settings.db` in case I break something, I just rename it back.

**Comment by samuel-w** [comment]
Additionally, Windows backup software does not have this issue. It just restores the data.
Also, the scenario I gave was just my scenario, clearly @aromanelli also has a use for this.

Going to repeat my point:  You can't "train" people by having Vorta crash, they'll just file another bug report and get annoyed at the software. 
In the end, all I want to prevent is the crash from overwriting settings.db through restoring a backup, any kind of backup. Vorta's whole point is to be a user friendly way to backup and restore data. This fulfills Vorta's purpose.

**Comment by m3nu** [comment]
Please accept that I don't want to merge such a feature. Arguing here is very tiring and I'd rather do something productive while working on this project.

**Comment by samuel-w** [comment]
Just remember, I am a volunteer

**Comment by aromanelli** [comment]
> Please accept that I don't want to merge such a feature. Arguing here is very tiring and I'd rather do something productive while working on this project.

No disrespect, realize you are a volunteer, but you are on the wrong side of this. The whole point of Vorta is not to just make a Linux nerd's job easier, its to allow people migrating over from Windows be able to use a powerful backup/restore engine in an easy-to-use way, so that its not impossible for them to use it.

Your justifications are not valid.  No developer should ever advocate for the purposeful crashing of the app and try to write it off as a training issue. To be blunt, that's just bad programming. Your program should always handle all edge-cases. Always.

Please reconsider, even if it means you have to "remember" the patch for any new work later down the line. That's part of the job description for a programmer, and I would hope that the person creating the patch would include documentation in-line to help out with any future bug fixes.

And finally, you really want to have Windows do something that Linux can't?  Really?

Apologies if the above seems harsh, but like I said, you are really on the wrong side of this and are just arguing so that you do not have to do work. If you were a paid employee, that would get you fired.  Since you are not paid, its totally in your sphere of control to decide what Vorta means and how it and you perceived by others.

Take care.

**Comment by Hofer-Julian** [comment]
Is today attack-the-maintainer-day?
You have no say in what Vorta's job is nor is it your decision what other people invest their free time in.

I know that you think you commented with best intentions, but if you want open source to improve, you'd be advised to stop.

**Comment by aromanelli** [comment]
> 
> 
> Is today attack-the-maintainer-day?
> You have no say in what Vorta's job is nor is it your decision what other people invest their free time in.
> 
> I know that you think you commented with best intentions, but if you want open source to improve, you'd be advised to stop.

We're not arguing to save our souls, but yours. Defending open source through bad decision making does not support open source. Its about making products that expands the usage of software to the masses, not justify enclosure by the elite.

NO programmer should justify the crashing of an app as a 'training issue'. Open or closed sourced.

**Pull request closed without merging**